### PR TITLE
ER-292, ER-293, ER-294, ER-295 Part 2 Validation

### PR DESCRIPTION
**Pull request by sachpatel**

### CODE REVIEW

File: src/Shared/Recruit.Vacancies.Client/Application/Validation/VacancyRuleSet.cs [hunk 1]
```diff
@@ -5,27 +5,33 @@ namespace Esfa.Recruit.Vacancies.Client.Application.Validation
...
+        Qualifications = 1 << 16,
+        Description = 1 << 17,
+        TrainingDescription = 1 << 18,
+        OutcomeDescription = 1 << 19,
+        ApplicationInstructions = 1 << 20,
+        ApplicationUrl = 1 << 21,
+        EmployerContactDetails = 1 << 22,
+        ThingsToConsider = 1 << 23,
+        EmployerDescription = 1 << 24,
+        EmployerWebsiteUrl = 1 << 25,
```
Comment (chrisjensenuk) [inline src/Shared/Recruit.Vacancies.Client/Application/Validation/VacancyRuleSet.cs]: 👍 I find this much easier to read.

File: src/Shared/UnitTests/Application/VacancyValidation/SingleField/EmployerContactDetailContactPhoneNumberTests.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            result.HasErrors.Should().BeFalse();
+            result.Errors.Should().HaveCount(0);
+        }
+
+        [Fact]
+        public void EmployerContactPhoneNumberMustBe16CharactersOrLess()
+        {
+            var vacancy = new Vacancy
+            {
+                EmployerContactPhone = "+4402086695847".PadRight(101, 'w')
```
Comment (chrisjensenuk) [inline src/Shared/UnitTests/Application/VacancyValidation/SingleField/EmployerContactDetailContactPhoneNumberTests.cs]: .PadRight(17, 'w')

File: src/Shared/UnitTests/Application/VacancyValidation/SingleField/EmployerWebsiteUrlValidationTests.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            };
+
+            var result = Validator.Validate(vacancy, VacancyRuleSet.EmployerWebsiteUrl);
+
+            result.HasErrors.Should().BeFalse();
+            result.Errors.Should().HaveCount(0);
+        }
+        
+        [Fact]
+        public void EmployerWebsiteUrlMustBe200CharactersOrLess()
```
Comment (chrisjensenuk) [inline src/Shared/UnitTests/Application/VacancyValidation/SingleField/EmployerWebsiteUrlValidationTests.cs]: EmployerWebsiteUrlMustBe100CharactersOrLess

File: src/Employer/Employer.Web/Mappings/TrainingProviderMapper.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                Address = GetAddress(provider.Addresses.FirstOrDefault(addr => addr.ContactType.Equals("PRIMARY")) ?? provider.Addresses.First())
+            };
+        }
+        
+        private static Address GetAddress(ContactAddress address)
+        {
+            return new Address
+            {
+                AddressLine1 = address.Primary,
+                AddressLine2 = address.Secondary,
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Mappings/TrainingProviderMapper.cs]: If address.Secondary is empty should we put in address.Street instead. And so on....

File: src/Employer/Employer.Web/Orchestrators/Part2/TrainingProviderOrchestrator.cs [hunk 5]
```diff
@@ -30,61 +36,80 @@ public TrainingProviderOrchestrator(IVacancyClient client, ITrainingProviderServ
...
+                    ProviderName = provider.Name,
+                    ProviderAddress = provider.Address.GetInlineAddress()
+                };
             }
             else
             {
-                confirmViewModel.Ukprn = vacancy.Ukprn.Value;
-                confirmViewModel.ProviderName = vacancy.ProviderName;
-                confirmViewModel.ProviderAddress = vacancy.ProviderAddress;
+                return new ConfirmTrainingProviderViewModel
```
Comment (chrisjensenuk) [inline src/Employer/Employer.Web/Orchestrators/Part2/TrainingProviderOrchestrator.cs]: Don't think you need the 'else' above

Comment (sachpatel) [reply src/Employer/Employer.Web/Orchestrators/Part2/TrainingProviderOrchestrator.cs]: 👍

File: src/Employer/Employer.Web/Mappings/TrainingProviderMapper.cs (hunk 4 above)
Comment (sachpatel) [reply src/Employer/Employer.Web/Mappings/TrainingProviderMapper.cs]: This is sorted on the provider API (I hope, I was the one that did the work 😬 ) when it gets its data from UKRLP. Will double check!

Comment (sachpatel) [reply src/Employer/Employer.Web/Mappings/TrainingProviderMapper.cs]: It should be okay since we are reducing when displaying the address.

**Pull request merged**